pkg/reexec/reexec.go

```diff
@@ -49,3 +54,13 @@ func naiveSelf() string {
 	// (NOTE: Go only errors on Abs() if os.Getwd fails)
 	return name
 }
+
+func WithOTelInterceptor(serviceName string, inner func(context.Context)) func() {
+	return func() {
+		ctx := otelutil.ExtractEnvVars(context.Background())
+		ctx, span := otel.Tracer("").Start(ctx, serviceName, trace.WithSpanKind(trace.SpanKindServer))
+		defer span.End()
+
+		inner(ctx)
+	}
+}
```

Comment:
As much as I like FP idioms such as decorators, it feels weird to me to decorate (effectively) a `func main()`. There's no way to perform [early initialization](https://pkg.go.dev/go.opentelemetry.io/otel#SetTextMapPropagator) aside from wrapping with another decorator. It works out okay because `otelutil.ExtractEnvVars` currently is hardcoded to use one particular propagator, but we'd have to make a (potentially breaking) change to the behaviour of a non-internal package if we wanted to do anything differently. I think we should simply inline this code into `func processSetKeyReexec` and avoid adding any OTEL propagation related stuff to any `pkg/...` packages if it can be avoided.